lib/finitomata/flow.ex

```diff
@@ -0,0 +1,557 @@
+defmodule Finitomata.Flow do
+  @moduledoc """
+  The basic “brick” to build forks in top-level `Finitomata` instances
+  """
+
+  alias Finitomata.Transition
+
+  @start_state "finitomata_flowing"
+  @start_event "finitomata_flow_initialize!"
+  @start_handler :on_flow_initialization
+  @end_state "finitomata_flowed"
+  @back_event "finitomata_back"
+
+  @doc false
+  defmacro __using__(opts \\ []) do
```

Comment:
would it be possible to split this into calls to defps? This function is super hard to read.